kratos/utilities/function_parser_utility.cpp

```diff
@@ -141,8 +153,67 @@ void BasicGenericFunctionUtility::InitializeParser()
         const te_variable vars[] = {{"x", &x}, {"y", &y}, {"z", &z}, {"t", &t}, {"X", &X}, {"Y", &Y}, {"Z", &Z}};
 
         /* Compile the expression with variables. */
-        mpTinyExpr = te_compile(mFunctionBody.c_str(), vars, 7, &err);
-        KRATOS_ERROR_IF_NOT(mpTinyExpr) << "Parsing error in function: " << mFunctionBody << std::endl;
+        const bool python_like_ternary = StringUtilities::ContainsPartialString(mFunctionBody, "if") ? true : false;
+        if (!python_like_ternary) {
+            mpTinyExpr[0] = te_compile(mFunctionBody.c_str(), vars, 7, &err);
+            KRATOS_ERROR_IF_NOT(mpTinyExpr[0]) << "Parsing error in function: " << mFunctionBody << std::endl;
+        } else { // Ternary operator
+            mpTinyExpr.resize(3, nullptr);
+            std::string condition, first_function, second_function;
+
+            // C like ternary operator
+            std::vector<std::string> splitted_string;
+            KRATOS_ERROR_IF_NOT(StringUtilities::ContainsPartialString(mFunctionBody, "else")) << "Parsing error in function: " << mFunctionBody << " if defined, but not else" << std::endl;
+            std::string aux_string = StringUtilities::ReplaceAllSubstrings(mFunctionBody, "if", "$");
```

Comment:
why is this necessary?

Reply:
Tinyexpr have some issues :P , that's why is auxiliar

Reply:
can you briefly explain please, for the record?